test/SimpleSwap.behavior.js

```diff
@@ -1,931 +1,1471 @@
-                      shouldSubmitChequeBeneficiary(unsignedCheque, sender)
+
+    describe(describeFunction + 'submitChequeIssuer', function () {
+      if (enabledTests.submitChequeIssuer) {
+        context("when we don't send value along", function () {
+          const value = new BN(0)
+          context('when the sender is the issuer', function () {
+            const sender = issuer
+            context('when the first serial is higher than 0', function () {
+              expect(defaultCheque.serial).bignumber.to.be.above(new BN(0), "Serial of defaultCheque not above 0")
+              context('when the first serial is below MAX_UINT256', function () {
+                expect(defaultCheque.serial).bignumber.to.be.below(constants.MAX_UINT256, "Serial of defaultCheque not above 0")
+                context('when the beneficiary is a signee', function () {
+                  const unsignedCheque = Object.assign({}, defaultCheque, { signee: defaultCheque.beneficiary })
+                  expect(unsignedCheque.signee).to.be.equal(unsignedCheque.beneficiary, "Signee of unsignedCheque is not beneficiary")
+                  context('when the signee signs the correct fields', function () {
+                    context('when we send one cheque', function () {
+                      context('when there is a liquidBalance to cover the cheque', function () {
+                        describe(describePreCondition + 'shouldDeposit', function () {
+                          describe(describePreCondition + 'shouldDeposit', function () {
+                            shouldDeposit(unsignedCheque.amount.add(unsignedCheque.amount), issuer)
+                            describe(describeTest + 'submitChequeIssuer', function () {
+                              shouldSubmitChequeIssuer(unsignedCheque, sender)
+                            })
+                          })
+                        })
+                      })
+                      context('when there is no liquidBalance to cover the cheque', function () {
+                        describe(describeTest + 'shouldSubmitChequeIssuer', function () {
+                          shouldSubmitChequeIssuer(unsignedCheque, sender)
+                        })
+                      })
                     })
-                    context('when there is no liquidBalance to cover the cheque', function() {
-                      shouldSubmitChequeBeneficiary(unsignedCheque, sender)  
+                    context('when we send more than one cheque', async function () {
+                      describe(describePreCondition + 'shouldSubmitChequeIssuer', function () {
+                        shouldSubmitChequeIssuer(unsignedCheque, sender)
+                        context('when the serial number is increasing', function () {
+                          describe(describeTest + 'shouldSubmitChequeIssuer', function () {
+                            const secondSerial = new BN(parseInt(unsignedCheque.serial) + 1)
```

Comment:
Why do we need to parse ints here?

Reply:
That is not needed. I will get rid of that